tests/integration/conftest.py

```diff
@@ -113,7 +116,9 @@ def factory(
         options: str,
         timeout: int = 60 * 60,
         unregister: bool = False,
+        expected_exitcode: Union[List[int], int] = 0,
```

Comment:
The idea of this accepting a list of expected return codes is that we can say for example: 

```python
expected_exitcode = [0, 1]
```

And the underlaying framework (pexpect `exitstatus`) will check if that return code is inside our list of expected, right?

If my understanding is right, I kinda have a few questions around this: 

1. Could we mess up tests by saying that `expected_exitcode = [0, 1, 2]` and not verify correctly that the tool is giving the real expected exit code? Of course, we have reviews and such, by I can also see this being used to avoid checking exit codes at all.
2. I haven't seen any case in the current commit that uses a list of exit codes to check the return (might be because I overlooked), but couldn't we start simpler and use just only one `expected_exitcode`?